src/dynsight/_internal/utilities/utilities.py

```diff
@@ -16,3 +19,42 @@ def normalize_array(x: np.ndarray) -> np.ndarray:  # type: ignore[type-arg]
     norm = np.linalg.norm(x, axis=-1, keepdims=True)
     norm[norm == 0] = 1
     return x / norm
+
+
+def find_extrema_points(
+    x_axis: np.ndarray[float, Any],
```

Comment:
You use the type "Any" a lot, which is very vague - it would be better to have it be something like `float |int|str|list` if it can be a lot of things. Can you try and fix this throughout?